src/overview.md

```diff
@@ -0,0 +1,126 @@
+the service worker intercepts the request, and then decides whether to serve a
+local version of the resource or to fetch a fresh version from the network.
+
+[proxy]: https://en.wikipedia.org/wiki/Proxy_server
+
+The place where Workbox stores resources locally is called the **cache**.
+The act of storing locally is called **caching**. This is because the
+underlying technology that Workbox relies on is called the [Cache API][cache].
+This is completely different than [HTTP caching][http]. The main benefit
+that Workbox provides over HTTP caching is that it can cache resources *before*
+they're requested, whereas HTTP caching can only do so after the user goes to
+a page where the resource is required.
+
+[cache]: https://developer.mozilla.org/en-US/docs/Web/API/Cache
+[http]: https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching
+
+The logic for determining whether to serve a resource from the cache or to
+fetch a fresh version is called the **caching strategy** for that resource.
+It's common to use different caching strategies for different types of
+resources. For example:
+
+* For a logo that doesn't change often, you can probably serve it from the
+  cache.
+* For a JSON file containing a user's social media feed, you probably want
+  to display the cached content only while you fetch the latest content.
+
+## What problems Workbox solves
+
+In short, Workbox makes it easy to create optimal service worker code.
+
+Manually maintaining caches is tedious and full of pitfalls.
+The "core" Workbox libraries and tools --- `workbox-sw`, `workbox-build`,
+`workbox-webpack-plugin`, and `workbox-cli` --- make it easy to automatically
+generate a service worker that correctly caches resources.
+
+If you need to use different caching strategies for different types of
+resources, Workbox also simplifies this process. Much of the logic can be
+reduced to declarative build-time configurations. The rest of it can be
+abstracted away by calling the `workbox-sw` library from your custom
+service worker code.
+
+Workbox also intelligently manages updates to cached resources. In other words,
+when a resource is changed, Workbox only downloads the changed resource and
+leaves the rest of the cache  untouched. This is a subtle but important
+optimization. Other solutions invalidate the entire cache when a single
+resource is updated, causing the service worker to re-download every resource.
+
+## How you use Workbox
+
+The general workflow for integrating Workbox into your project is:
+
+1. Create a configuration file telling Workbox which resources to cache.
+1. Run one of the Workbox tools in your project directory before each deploy.
+   Workbox generates the service worker code needed for caching and updating
+   the resources.
+
+There are a few Workbox tools, each one built for a different workflow.
+`workbox-webpack-plugin` is for webpack, `workbox-build` for Gulp, and
```

Comment:
`workbox-build` is a generic JS module, which could be used from Gulp, but could also be used from other things (Grunt, Broccoli, your-random-JS-build-system).

You can rephrase as:

> `workbox-build` for any node-based build tool, like Gulp or Grunt,